online-fishing-store/src/app/pages/main-page/main-page.component.ts

```diff
@@ -42,6 +58,7 @@ export class MainPageComponent implements OnInit {
   }
 
   ngOnInit(): void {
+    this.getProducts()
     this.products = this.sortServise.sortByFavorites(this.products);
```

Comment:
In case if you need to show always sorted products by Favorites you can get here already sorted array. You can add this sorting in products service in getProducts() method.

You can use for it "map" operator or "tap"